pkg/model/config/config.go

```diff
@@ -147,44 +220,84 @@ func (c Config) resourceAPIVersionCompatible(verType, version string) bool {
 	return true
 }
 
-// GVK contains information about scaffolded resources
-type GVK struct {
-	Group   string `json:"group,omitempty"`
-	Version string `json:"version,omitempty"`
-	Kind    string `json:"kind,omitempty"`
+// simplify omits some fields that can be restored at unmarshalling.
+func (c *Config) simplify() {
+	if c.IsV3() {
+		for i, r := range c.Resources {
+			// If the plural is regular, omit it.
+			if r.Plural == resource.RegularPlural(r.Kind) {
+				c.Resources[i].Plural = ""
+			}
+			// If the path is the default location, omit it.
+			if r.Path == resource.LocalPath(c.Repo, r.Group, r.Version, c.MultiGroup) {
+				c.Resources[i].Path = ""
+			}
+			// If API is empty, omit it (prevents `api: {}`).
+			if r.API != nil && r.API.IsEmpty() {
+				c.Resources[i].API = nil
+			}
+			// If Webhooks is empty, omit it (prevents `webhooks: {}`).
+			if r.Webhooks != nil && r.Webhooks.IsEmpty() {
+				c.Resources[i].Webhooks = nil
+			}
+		}
+	}
+}
 
-	// CRDVersion holds the CustomResourceDefinition API version used for the GVK.
-	CRDVersion string `json:"crdVersion,omitempty"`
-	// WebhookVersion holds the {Validating,Mutating}WebhookConfiguration API version used for the GVK.
-	WebhookVersion string `json:"webhookVersion,omitempty"`
+// restore sets the omitted values that were simplified during marshalling.
+func (c *Config) restore() {
+	if c.IsV3() {
+		for i, r := range c.Resources {
+			// Restore regular plural that were omitted.
+			if r.Plural == "" {
+				c.Resources[i].Plural = resource.RegularPlural(r.Kind)
+			}
+			// Restore the default location that was omitted.
+			if r.Path == "" {
+				c.Resources[i].Path = resource.LocalPath(c.Repo, r.Group, r.Version, c.MultiGroup)
+			}
+			// Create a pointer to an empty struct instead of nil to prevent panics.
+			if r.API == nil {
+				c.Resources[i].API = &resource.API{}
+			}
+			// Create a pointer to an empty struct instead of nil to prevent panics.
+			if r.Webhooks == nil {
+				c.Resources[i].Webhooks = &resource.Webhooks{}
+			}
```

Comment:
```suggestion
```
I am as maintainer would expected be able to check if a resource has an api or webhooks scafolled by doing `if resouce.API != nil or resource.webhook != nil` since this atttrs are pointers.

Reply:
API can't be a pointer because it will yield errors when templating.

[Example](https://play.golang.org/p/GKqt4c2ayWs)